### PR TITLE
added some tests

### PR DESCRIPTION
**Pull request by 5saviahv**
added/changed some tests so more conditions would be covered

### CODE REVIEW

File: lib/api/traversing.js [hunk 1]
```diff
@@ -697,7 +697,7 @@ exports.not = function (match, container) {
     });
   }
 
-  return container._make(elements);
```
Comment (fb55) [inline lib/api/traversing.js:700]: This is using the same pattern as `filter` right now. `filter` can be called with `this` being an array. We don't do that anywhere for `not` afaict, but maintaining the same pattern should be good in any case.

**Comment by fb55** [comment]
This looks great! The change in `.not` reduces consistency and should probably be reverted, otherwise these are some great changes.

File: lib/api/traversing.js (hunk 1 above)
Comment (5saviahv) [reply lib/api/traversing.js:700]: ok, reverted

**Comment by fb55** [comment]
Thanks for following up, and nice patch!

**Pull request merged**